test/abstractarray.jl

```diff
@@ -598,6 +598,22 @@ function test_get(::Type{TestAbstractArray})
     @test get(TSlow([]), (), 0) == 0
     @test get(TSlow([1]), (), 0) == 1
     @test get(TSlow(fill(1)), (), 0) == 1
+
+    @test get(()->0, A, ()) == 0
```

Comment:
Could we add some tests with a function that has side effects to test that this function only gets called if the index doesn't exist?